sw/host/ot_certs/src/x509/extension.rs

```diff
@@ -94,6 +134,52 @@ pub struct DiceTcbInfo<'a> {
     pub tcb_type: Option<&'a [u8]>,
 }
 
+fn asn1utf8_to_str(s: &asn1::Utf8String) -> Value<String> {
+    Value::literal(s.as_str().to_string())
+}
+
+fn asn1bigint_to_bn(bn: &asn1::BigInt) -> Value<BigUint> {
+    Value::literal(BigUint::from_bytes_be(bn.as_bytes()))
+}
+
+impl DiceTcbInfo<'_> {
```

Comment:
Out of curiosity, why are there two DiceTcbInfo structs (one in `extension.rs` and one in `dice_tcb.rs` and an additional `DiceExtension` struct in `template/mod.rs`? Can we not collapse these into one struct representation within this lib and impl the appropriate methods on it to instantiate it from asn1 and conversely gen asn1 from it?

Reply:
The one in `extension.rs` cannot be removed because it is the one used to derive the parser so it contains stuff like tag and asn1 specific marking. It really ought to be private really (might change that in that PR). The other two will be merged.